jujugui/static/gui/src/app/components/user-profile/user-profile.js

```diff
@@ -150,8 +151,267 @@ YUI.add('user-profile', function() {
       }
     },
 
+    /**
+      Generate the rows of model details.
+
+      @method _generateModelRows
+      @returns {Array} The markup for the rows.
+    */
+    _generateModelRows: function() {
+      var components = [];
+      this.state.envList.forEach((model) => {
+        var uuid = model.uuid;
+        components.push(
+          <juju.components.UserProfileEntity
```

Comment:
Instead of using some kind of magic component with a whitelist of fields and some kind of definition for all the different labels and column widths etc. I ended up going with this wrapper component. It means a little more duplication, but it was a lot easier to build, and hopefully therefore to maintain.